### ISSUE TITLE
Multipe rows with gutter containing responsive width columns do not are not spaced properly

### ISSUE THREAD

**Issue by simonsankar** [ant-design/ant-design#21282]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/condescending-grass-rd5lw?fontsize=14&hidenavigation=1&theme=dark)

### Steps to reproduce
1. Create a row and set the gutter to an array - gutter={[16,16]}
2. Add columns with responsive widths - xxl={3} lg={4} md={6} sm={8} xs={8}
3. Repeat the Row containing the columns multiple times
4. Resize the browser to see how the columns of the second row begin after the last column of the first row

### What is expected?
It is expected that each row's columns start at the leftmost end

### What is actually happening?
After the first row's columns end, the second row's columns start below as expected on the Y-axis but are not to the leftmost end on the X-axis but after the first row's last column

| Environment | Info |
|---|---|
| antd | 3.26.7 |
| React | reactjs |
| System | Windows10 |
| Browser | chrome |

**Comment by zombieJ** [comment]
Revert patch. Seems not find better solution.

**Comment by zombieJ** [comment]
Seems user can rewrite Row display style. Patch fix makes more [style issue](#21403). You can use `type=flex` for fixing this.

And in v4, this is not a issue more since it default use flex layout.

https://codesandbox.io/s/intelligent-kapitsa-31ybi

**Comment by felipeptcho** [comment]
The clearfix is not working because we have negative margin-top and margin-bottom on the grid rows when using vertical gutter. Instead of setting `clear: both` just for the `::after` pseudo-element, we should set it also for the `::before` one.

```diff
// mixins for clearfix
// ------------------------
.clearfix() {
  zoom: 1;
  &::before,
  &::after {
    display: table;
    content: '';
+   clear: both;
  }
-  &::after {
-    clear: both;
-  }
}
```

I tested this and it will not impact the other related ticket.
